Setup/UpgradeData.php

```diff
@@ -158,4 +162,58 @@ public function updateSchemaVersion244(ModuleDataSetupInterface $setup)
         // re-initialize otherwise it will cause errors
         $this->reinitableConfig->reinit();
     }
-}
\ No newline at end of file
+
+    /**
+     * Upgrade to 6.0.0
+     * The new configuration UI uses new fields, combines selections that used
+     * to be separated and splits old fields into multiple configurations
+     *
+     * @param ModuleDataSetupInterface $setup
+     */
+    public function updateDataVersion600(ModuleDataSetupInterface $setup)
+    {
+
+        $pathTerminalSelection = "payment/adyen_pos_cloud/terminal_selection";
+        $pathPosStoreId = "payment/adyen_pos_cloud/pos_store_id";
+
+        $configDataTable = $setup->getTable('core_config_data');
+        $connection = $setup->getConnection();
+
+        //Setting terminal_selection=store_level for scopes that have pos_store_id set
+        $select = $connection->select()
+            ->from($configDataTable)
+            ->where('path = ?', $pathPosStoreId)
+            ->where('value <> "" AND value IS NOT NULL');
+        $configPosStoreIdValues = $connection->fetchAll($select);
+        foreach ($configPosStoreIdValues as $configPosStoreIdValue) {
+            $scope = $configPosStoreIdValue['scope'];
+            $scopeId = $configPosStoreIdValue['scope_id'];
+            $this->configWriter->save(
+                $pathTerminalSelection,
+                'store_level',
+                $scope,
+                $scopeId
```

Comment:
just wondering if we take into account that a setting can be in multiple stores. Do we include storeId into the read an save operation ?

Reply:
Correct, the $scope and $scopeId variables localize the config value to website, store or storeview. If there's a pos_store_id setting for a specific storeview we'll apply the terminal_selection in that specific scope as well.